freedomofpress/securedrop
# Modify CI docker SD behavior to match dev env

## Status

Ready for review -- uncovered what look like legit linting failures

## Description of Changes

Fixes #3839 

Changes proposed in this pull request:

* Utilize higher :dollar:  cost `machine` executor (at least they plan to charge more for it in the "future")
* Revamp docker caching to only rebuild when `Dockerfile` changes.

## Testing

How should the reviewer test this PR?

* Ensure you can still spin up docker locally (tests and dev env)
* Ensure CI doesn't break :crossed_fingers: 

## Deployment

Any special considerations for deployment? Consider both:

NOPE - only affects dev + CI

## Checklist

### If you made changes to the server application code:

Umm  I didn't questionnaire... why you all up in my bizness!?

### If you made changes to `securedrop-admin`:

HEY! Whats with all the :fire: ?

### If you made changes to the system configuration:

I swear to god... 

### If you made non-trivial code changes:


### If you made changes to documentation:

Docs!? we have docs!?
